pkgs/by-name/mp/mp3blaster/package.nix

```diff
@@ -8,15 +8,15 @@
   SDL,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "mp3blaster";
   version = "3.2.6";
 
   src = fetchFromGitHub {
     owner = "stragulus";
     repo = "mp3blaster";
-    rev = "v${version}";
-    sha256 = "0pzwml3yhysn8vyffw9q9p9rs8gixqkmg4n715vm23ib6wxbliqs";
+    tag = "v${finalAttrs.version}";
```

Comment:
I've seen this pattern a few times now, but I don't understand it.  If the version comes from `finalAttrs`, then isn't the `version` variable up on line 13 unused?  I don't quite follow how passing a function to `mkDerivation` works, or where `finalAttrs` comes from.

Based on the explanation [here](https://ryantm.github.io/nixpkgs/stdenv/stdenv/#mkderivation-recursive-attributes), I think `finalAttrs.version` references whatever value is set on line 13, but I'm not sure.

Reply:
> Based on the explanation [here](https://ryantm.github.io/nixpkgs/stdenv/stdenv/#mkderivation-recursive-attributes), I think `finalAttrs.version` references whatever value is set on line 13

Yes

Reply:
More details https://nixos.org/manual/nixpkgs/stable/#mkderivation-recursive-attributes